pkg/cluster/install.go

```diff
@@ -21,46 +21,82 @@ import (
+		)
+	}
+
+	// Make sure the VMs are switched on and we have an APIServer -- fail if we
+	// don't, because we can't do the following steps
+	toRun = append(toRun,
 		steps.Action(m.startVMs),
-		steps.Condition(m.apiServersReady, 30*time.Minute, false),
-		steps.Action(m.ensureBillingRecord), // belt and braces
-		steps.Action(m.configureAPIServerCertificate),
-		steps.Action(m.configureIngressCertificate),
-		steps.Action(m.fixSSH),
-		steps.Action(m.fixInfraID),        // Old clusters lacks infraID in the database. Which makes code prone to errors.
-		steps.Action(m.populateCreatedAt), // TODO(mikalai): Remove after a round of admin updates
-		steps.Action(m.fixSREKubeconfig),
-		steps.Action(m.fixUserAdminKubeconfig),
-		steps.Action(m.createOrUpdateRouterIPFromCluster),
-		steps.Action(m.populateDatabaseIntIP),
-		steps.Action(m.fixMCSCert),
-		steps.Action(m.fixMCSUserData),
-		steps.Action(m.ensureGatewayUpgrade),
-		steps.Action(m.ensureAROOperator),
-		steps.Condition(m.aroDeploymentReady, 20*time.Minute, false),
-		//steps.Action(m.removePrivateDNSZone), // TODO(mj): re-enable once we communiate this out
-		steps.Action(m.updateProvisionedBy), // Run this last so we capture the resource provider only once the upgrade has been fully performed
+		steps.Condition(m.apiServersReady, m.applyTimeDilation(30*time.Minute), true),
+	)
+
+	// Requires Kubernetes clients
+	if isEverything {
+		toRun = append(toRun,
+			steps.NonCriticalAction(m.fixSREKubeconfig),
+			steps.NonCriticalAction(m.fixUserAdminKubeconfig),
+			steps.NonCriticalAction(m.createOrUpdateRouterIPFromCluster),
+			steps.NonCriticalAction(m.fixMCSCert),
+			steps.NonCriticalAction(m.fixMCSUserData),
+			steps.NonCriticalAction(m.ensureGatewayUpgrade),
+			steps.NonCriticalAction(m.configureAPIServerCertificate),
+			steps.NonCriticalAction(m.configureIngressCertificate),
+			steps.NonCriticalAction(m.populateCreatedAt), // TODO(mikalai): Remove after a round of admin updates
+
+		)
 	}
 
-	return m.runSteps(ctx, steps)
+	// Update the ARO Operator
+	if isEverything || isOperator {
+		toRun = append(toRun,
+			steps.Action(m.ensureAROOperator),
+			// Don't let the Operator fail to roll out pass silently
+			steps.Condition(m.aroDeploymentReady, m.applyTimeDilation(20*time.Minute), true),
+		)
+	}
+
+	// We don't run this on an operator-only deploy as PUCM scripts then cannot
+	// determine if the cluster has been fully admin-updated
+	if isEverything {
+		toRun = append(toRun,
+			steps.Action(m.updateProvisionedBy), // Run this last so we capture the resource provider only once the upgrade has been fully performed
+		)
+	}
+
+	return m.runSteps(ctx, toRun)
 }
 
 func (m *manager) Update(ctx context.Context) error {
 	steps := []steps.Step{
-		steps.AuthorizationRefreshingAction(m.fpAuthorizer, steps.Action(m.validateResources)),
+		steps.AuthorizationRefreshingAction(m.fpAuthorizer, m.applyTimeDilation(10*time.Minute), steps.Action(m.validateResources)),
```

Comment:
Why do we add timeDilatation? Again, does not look like part of this PR but might be wrong

Reply:
Re-thought how to do this, fixed